src/Realizer.js

```diff
@@ -8,9 +8,38 @@ class Realizer
 		this.isFullyUpdated = false;
 		this.moveQueue = [];
 		
+		this.activeCell = -1;
+		this.activeCellCanMove = [];
+		this.activeCellCanCapture = [];
+		
 		this.cellsPerRow = Math.round(Math.sqrt(this.board.contents.length));
 	}
 	
+	SetActiveCell(newActiveCell)
+	{
+		this.activeCell = newActiveCell;
+		this.activeCellCanMove = [];
+		this.activeCellCanCapture = [];
+		
+		if (this.activeCell > 0 && this.board.contents[this.activeCell] !== undefined)
+		{
+			const activePiece = this.board.contents[this.activeCell];
+			
+			activePiece.moveVectors.forEach((vector) => {
+				this.activeCellCanMove = this.activeCellCanMove.concat(this.board.GetCellIndices(vector, this.activeCell, false));
```

Comment:
Prefer to do `this.activeCellCanMove.push(...this.board.GetCellIndices(vector, this.activeCell, false))` for this and the other calls to `concat`.

This makes `this.activeCellCanMove` immutable which helps testing.